File: hayden/2023/d1/e2.go
package main

import (
	"os"
	"strconv"
	"strings"
)

func main() {
	input, err := os.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	lines := strings.Split(string(input), "\n")
	acc := 0
	for _, line := range lines {
		//Forward Lookup
		var numbers []int
		for i := 0; i < len(line); i++ {
			char := line[i]
			digit, err := strconv.Atoi(string(char))
			//fast out
			if err == nil {
				numbers = append(numbers, digit)
				continue
			}
			if i+3 <= len(line) {
				lah3 := line[i : i+3]
				switch lah3 {
				case "one":
					numbers = append(numbers, 1)
					continue
				case "two":
					numbers = append(numbers, 2)
					continue
				case "six":
					numbers = append(numbers, 6)
					continue
				}
			}
			if i+4 <= len(line) {
				lah4 := line[i : i+4]
				switch lah4 {
				case "four":
					numbers = append(numbers, 4)
					continue
				case "five":
					numbers = append(numbers, 5)
					continue
				case "nine":
					numbers = append(numbers, 9)
					continue
				}
			}
			if i+5 <= len(line) {
				lah5 := line[i : i+5]
				switch lah5 {
				case "three":
					numbers = append(numbers, 3)
					continue
				case "seven":
					numbers = append(numbers, 7)
					continue
				case "eight":
					numbers = append(numbers, 8)
					continue
				}
			}
		}
		acc += numbers[0]*10 + numbers[len(numbers)-1]
	}
	println(acc)
}

File: hayden/2023/d1/e2.js
const fs = require('fs');

const file = fs.readFileSync('input.txt', 'utf-8')
const lines = file.split("\n");

let lineSum = 0;
//All my Top -> Bottom Work
for (const line of lines){
    let foundNumbers = [];
    //All my Left -> Right Work
    for (let index = 0; index < line.length; index++ ){
        const char = line[index]
        let testNumber = Number(char);
        if (! Number.isNaN(testNumber)){
            foundNumbers.push(testNumber);
        } else {
            //I am going to look at the next 4 characters to see if a word is spelled
            const lookAhead = line.substring(index, index+5)
            if (lookAhead.indexOf("one") == 0) {
                foundNumbers.push(1)
            } else if (lookAhead.indexOf("two") == 0 ) {
                foundNumbers.push(2)
            } else if (lookAhead.indexOf("three") == 0){
                foundNumbers.push(3)
            } else if (lookAhead.indexOf("four") == 0){
                foundNumbers.push(4)
            } else if (lookAhead.indexOf("five") == 0){
                foundNumbers.push(5)
            } else if (lookAhead.indexOf("six") == 0){
                foundNumbers.push(6)
            } else if (lookAhead.indexOf("seven") == 0){
                foundNumbers.push(7)
            } else if (lookAhead.indexOf("eight") == 0){
                foundNumbers.push(8)
            } else if (lookAhead.indexOf("nine") == 0){
                foundNumbers.push(9)
            }
        }
    }
    //Work that needs all the L->R work done Before Going Down to Next Line
    lineSum += foundNumbers[0] * 10 + foundNumbers[foundNumbers.length-1]
}
//Work that needs all the T -> B Work Done
console.log(lineSum)